realm/realm-library/src/main/cpp/jni_util/java_local_ref.hpp

```diff
@@ -41,15 +41,26 @@ class JavaLocalRef {
     inline JavaLocalRef(JNIEnv* env, T obj, NeedToCreateLocalRef) noexcept
         : m_jobject(env->NewLocalRef(obj))
         , m_env(env){};
+
     inline ~JavaLocalRef()
     {
         m_env->DeleteLocalRef(m_jobject);
     }
 
-    JavaLocalRef(const JavaLocalRef&) = delete;
-    JavaLocalRef& operator=(const JavaLocalRef&) = delete;
-    JavaLocalRef(JavaLocalRef&& rhs) = delete;
-    JavaLocalRef& operator=(JavaLocalRef&& rhs) = delete;
+    JavaLocalRef& operator=(JavaLocalRef&& rhs)
+    {
+        this->~JavaLocalRef();
+        new (this) JavaLocalRef(std::move(rhs));
+        return *this;
+    }
+
+    inline JavaLocalRef(JavaLocalRef&& rhs)
+    {
+        m_env = rhs.m_env;
+        m_jobject = (rhs.m_jobject != nullptr) ? m_env->NewLocalRef(rhs.m_jobject) : nullptr;
```

Comment:
This will leak `rhs.m_jobject`. well, in this special case, it may not leak the local ref since all the local ref will be released when jni return to java. But in the C++ wise, this create a new local ref which means there are two local refs to the same object, and then the `rhs.m_jobject` is set to null, which is a leak.

Won't code below work?
```c++
inline JavaLocalRef(JavaLocalRef&& rhs) 
    : m_env(rhs.m_env), m_jobject(rhs.m_jobject)
{
    m_jobject = nullptr;
}
```

initialization list is preferred since it will avoid setting m_jojbect twice.